### PR TITLE
[feat]: [CDS-47037] Asg NG Blue Green Swap Service Delegate  Changes 

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
Asg NG Blue Green Swap Service Delegate  Changes 
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42712)

### CODE REVIEW

**Review by acasian-harness — CHANGES_REQUESTED** [review]
> Reviewed 23 of 23 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42712)* status: 23 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java` line 192 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkJ73DaexWTrPbY1S:-NLzkJ73DaexWTrPbY1T:b-z7bs20) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java#L192)):*
> ```Java
>             .loadBalancer("lovish-alb")
>             .prodListenerArn(
>                 "arn:aws:elasticloadbalancing:us-east-1:806630305776:listener/app/lovish-alb/4ff4788028fe646f/13c2c1652e75e445")
> ```

why are we hardcoding this? if it'a draft pr - please mark it as draft, we cannot have hardcoded things merged in develop.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java` line 209 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkUVTClHziZ2cJDUM:-NLzkUVTClHziZ2cJDUN:bb7od0r) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java#L209)):*
> ```Java
>             .timeoutIntervalInMin(CDStepHelper.getTimeoutInMin(stepParameters))
>             .asgLoadBalancerConfig(asgLoadBalancerConfig)
>             .oldServiceName("lovish2-asg")
> ```

same
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 164 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkzsa-3EH0LauK7-a:-NLzkzsa-3EH0LauK7-b:b-5q6i64) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L164)):*
> ```Java
>               .stageTargetGroupArnList(
>                   asgBlueGreenSwapServiceRequest.getAsgLoadBalancerConfig().getStageTargetGroupArnsList())
>               .trafficShifted(true)
> ```

why are always setting to true?
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 181 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzlOrk6e7g_3IRmGvo:-NLzlOrlCty-wQwzOvBJ:bjg446d) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L181)):*
> ```Java
>     } catch (Exception e) {
>       swapServiceLogCallback.saveExecutionLog(color(format("Swapping Failed. %n"), LogColor.Red, LogWeight.Bold),
>           LogLevel.ERROR, CommandExecutionStatus.FAILURE);
> ```

hint with explanation - according to new error handler framework.
___
*[`950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java` line 443 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzm9ZH9uIau_xVJmH9:-NLzm9ZI-Kr0adaAIs_K:by3kkjx) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java#L443)):*
> ```Java
>   TERRAFORM_INPUT_VARIABLES_OBTAIN_TASK_V2(TaskGroup.TERRAFORM, "Terraform Fetch Input variables Task V2"),
>   TERRAFORM_FETCH_TARGETS_TASK_V2(TaskGroup.TERRAFORM, "Terraform Fetch Targets Task V2"),
> ```

what is the change here?
___
*[`960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java` line 672 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzmmCSAbpMmNR6jYHP:-NLzmmCSAbpMmNR6jYHQ:bzcxuuv) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java#L672)):*
> ```Java
>       }
>       nextToken = describeRulesResponse.nextMarker();
>     } while (nextToken != null);
> ```

if it will be a problem on AWS side and token will be always null we will end up in infinite loop, we need to have a better strategy based on retry attempts with backoff strategy in place rather then firing too many requests and reaching aws request limit.
___
*[`960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java` line 694 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLznS7F8N_xuLQmSyJV:-NLznS7F8N_xuLQmSyJW:bwf990a) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java#L694)):*
> ```Java
>     newTag.withKey(BG_VERSION)
>         .withValue(newTagValue)
>         .withPropagateAtLaunch(true)
> ```

what is this property about?
___
*[`960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java` line 696 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLznOr_6LDNa7yK9cMq:-NLznOra3uejh-Zulz3J:bqsw4o9) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java#L696)):*
> ```Java
>         .withPropagateAtLaunch(true)
>         .withResourceId(asgName)
>         .withResourceType("auto-scaling-group");
> ```

constant
___
*[`960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java` line 754 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzonnf3wDzeWECvNEt:-NLzonnf3wDzeWECvNEu:b-2guby3) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java#L754)):*
> ```Java
>       listeners.addAll(describeListenersResponse.listeners());
>       nextToken = describeLoadBalancersResponse.nextMarker();
>     } while (nextToken != null);
> ```

same here

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42712)* status: 23 file(s) to review. 1 change requests. 8. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java` line 192 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkJ73DaexWTrPbY1S:-NLzqrlWEpZXyshLoXi8:b5hmkef) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java#L192)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

why are we hardcoding this? if it'a draft pr - please mark it as draft, we cannot have hardcoded things merged in develop.
</blockquote></details>

yes, removing that in next commit 
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java` line 209 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkUVTClHziZ2cJDUM:-NLzqyhu3yDwRY12ZKvb:b5hmkef) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java#L209)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

same
</blockquote></details>

yes, removing that in next commit 
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 164 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkzsa-3EH0LauK7-a:-NLzrZfbBVahP7SMY3u7:bg2wlkx) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L164)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

why are always setting to true?
</blockquote></details>

if swapping fails, this will remain by default false. 
___
*[`950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java` line 443 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzm9ZH9uIau_xVJmH9:-NLzryjL1V1eXsGs-yiN:b-nos4su) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java#L443)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

what is the change here?
</blockquote></details>

added spaces. removing them in next commit
___
*[`960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java` line 672 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzmmCSAbpMmNR6jYHP:-NLzsNqfEwVwFuBxBoFw:bawt97b) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java#L672)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

if it will be a problem on AWS side and token will be always null we will end up in infinite loop, we need to have a better strategy based on retry attempts with backoff strategy in place rather then firing too many requests and reaching aws request limit.
</blockquote></details>

okay, we can take it as a task later on ? as it is already same way implemented for ecs and elastigroup. 
___
*[`960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java` line 694 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLznS7F8N_xuLQmSyJV:-NLzsUe685vCPnkL__jK:b-dg9sy9) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java#L694)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

what is this property about?
</blockquote></details>

Some AWS Properties - required to be passed. But it wont cause any issue. 
___
*[`960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java` line 696 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLznOr_6LDNa7yK9cMq:-NLzs_LKDPX8-3aUWI3T:b-jjc181) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java#L696)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

constant
</blockquote></details>


Some AWS Properties - required to be passed. But it wont cause any issue. 
___
*[`960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java` line 754 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzonnf3wDzeWECvNEt:-NLzslb11J_h99mgmcQF:bawt97b) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/960-api-services/src/main/java/io/harness/aws/asg/AsgSdkManager.java#L754)):*
<details><summary><i>Previously, acasian-harness (Alexandru Casian) wrote…</i></summary><blockquote>

same here
</blockquote></details>

okay, we can take it as a task later on ? as it is already same way implemented for ecs and elastigroup.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java [hunk 1]
```diff
@@ -179,21 +182,45 @@ public TaskRequest obtainTaskAfterRbac(
...
+            .asgLoadBalancerConfig(asgLoadBalancerConfig)
+            .oldServiceName("lovish2-asg")
+            .newServiceName("lovish1-asg")
             //.oldServiceName(asgBlueGreenPrepareRollbackDataOutcome.getServiceName())
             //.newServiceName(asgBlueGreenDeployDataOutcome.getServiceName())
             .downsizeOldAsg(asgBlueGreenSwapServiceStepParameters.getDownsizeOldAsg().getValue() != null
                 && asgBlueGreenSwapServiceStepParameters.getDownsizeOldAsg().getValue())
             .build();
 
+    AsgBlueGreenSwapServiceStartOutcome asgBlueGreenSwapServiceStartOutcome =
```
Comment (vitsafronovici) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java:217]: why we need this `Start`

File: 960-api-services/src/main/java/io/harness/aws/asg/manifest/AsgSwapServiceHandler.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        awsInternalConfig);
+    asgSdkManager.info(color(format("Successfully updated Stage Listener %n%n"), LogColor.White, Bold), LogLevel.INFO);
+  }
+
+  public void modifyListenerRule(String region, String listenerArn, String listenerRuleArn,
+      List<String> targetGroupArnsList, AwsInternalConfig awsInternalConfig) {
+    // check if listener rule is default one in listener
+    if (asgSdkManager.checkForDefaultRule(region, listenerArn, listenerRuleArn, awsInternalConfig)) {
+      asgSdkManager.info(
+          format(
```
Comment (vitsafronovici) [inline 960-api-services/src/main/java/io/harness/aws/asg/manifest/AsgSwapServiceHandler.java:96]: info already has format

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java [hunk 3]
```diff
@@ -120,6 +120,7 @@ private StepResponse generateStepResponse(Ambiance ambiance,
               .stageListenerArn(asgBlueGreenSwapServiceResult.getStageListenerArn())
               .stageListenerRuleArn(asgBlueGreenSwapServiceResult.getStageListenerRuleArn())
               .stageTargetGroupArnList(asgBlueGreenSwapServiceResult.getStageTargetGroupArnList())
```
Comment (vitsafronovici) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java:122]: why we need all these fields if they are alreay in loadBalancer object

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceOutcome.java [hunk 4]
```diff
@@ -34,4 +34,5 @@ public class AsgBlueGreenSwapServiceOutcome implements Outcome, ExecutionSweepin
   String stageListenerArn;
   String stageListenerRuleArn;
   List<String> stageTargetGroupArnList;
```
Comment (vitsafronovici) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceOutcome.java:36]: why we need these fields, they are part of prepare rollback outcome

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java [hunk 5]
```diff
@@ -159,51 +160,53 @@ public TaskRequest obtainTaskAfterRbac(
...
             .commandName(ASG_BLUE_GREEN_SWAP_SERVICE_COMMAND_NAME)
             .commandUnitsProgress(CommandUnitsProgress.builder().build())
             .asgInfraConfig(asgStepCommonHelper.getAsgInfraConfig(infrastructureOutcome, ambiance))
             .timeoutIntervalInMin(CDStepHelper.getTimeoutInMin(stepParameters))
-            //.asgLoadBalancerConfig(asgLoadBalancerConfig)
-            //.oldServiceName(asgBlueGreenPrepareRollbackDataOutcome.getServiceName())
-            //.newServiceName(asgBlueGreenDeployDataOutcome.getServiceName())
+            .asgLoadBalancerConfig(asgLoadBalancerConfig)
+
+            .oldAsgName(asgBlueGreenPrepareRollbackDataOutcome.getAsgName())
```
Comment (vitsafronovici) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java:193]: should be prodAsgName

File: 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+            put(AsgConfigurationManifestHandler.OverrideProperties.maxSize, 0);
+            put(AsgConfigurationManifestHandler.OverrideProperties.desiredCapacity, 0);
+          }
+        };
+        List<String> asgScalingPolicyContentList = null;
+
+        chainState = AsgManifestHandlerChainFactory.builder()
+                         .initialChainState(AsgManifestHandlerChainState.builder()
+                                                .asgName(asgBlueGreenSwapServiceRequest.getOldAsgName())
+                                                .prodAsgName(asgBlueGreenSwapServiceRequest.getNewAsgName())
```
Comment (vitsafronovici) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java:113]: why prodasg is newName?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+                                 .build())
+
+                         .executeUpsert();
+      } else {
+        // Chain factory code to handle each manifest one by one in a chain
+        chainState = AsgManifestHandlerChainFactory.builder()
+                         .initialChainState(AsgManifestHandlerChainState.builder()
+                                                .asgName(asgBlueGreenSwapServiceRequest.getOldAsgName())
+                                                .prodAsgName(asgBlueGreenSwapServiceRequest.getNewAsgName())
+                                                .build())
```
Comment (vitsafronovici) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java:138]: why pros is new new name?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+                                 .region(asgInfraConfig.getRegion())
+                                 .awsInternalConfig(awsInternalConfig)
+                                 .build())
+                         .executeUpsert();
+      }
+
+      AsgBlueGreenSwapServiceResult asgBlueGreenSwapServiceResult =
+          AsgBlueGreenSwapServiceResult.builder()
+              .loadBalancer(asgBlueGreenSwapServiceRequest.getAsgLoadBalancerConfig().getLoadBalancer())
+              .prodTargetGroupArnList(
```
Comment (vitsafronovici) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java:153]: no need to pass back what you can get on manager side from Outcomes. And no need to put then in request if it is not needed on delegate side. Why AsgBlueGreenSwapServiceResult has so many repeating properties, if it's inside LoadBalancer object

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/asg/AsgBlueGreenSwapServiceResult.java [hunk 9]
```diff
@@ -26,4 +26,5 @@ public class AsgBlueGreenSwapServiceResult {
   private String stageListenerArn;
   private String stageListenerRuleArn;
   private List<String> stageTargetGroupArnList;
```
Comment (vitsafronovici) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/asg/AsgBlueGreenSwapServiceResult.java:28]: no need of these props

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceOutcome.java (hunk 4 above)
Comment (lovish1999) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceOutcome.java:36]: Hmm, removing them. But still we need trafficshifted boolean

File: 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+                  asgBlueGreenSwapServiceRequest.getAsgLoadBalancerConfig().getProdTargetGroupArnsList())
+              .prodListenerArn(asgBlueGreenSwapServiceRequest.getAsgLoadBalancerConfig().getProdListenerArn())
+              .prodListenerRuleArn(asgBlueGreenSwapServiceRequest.getAsgLoadBalancerConfig().getProdListenerRuleArn())
+              .stageListenerArn(asgBlueGreenSwapServiceRequest.getAsgLoadBalancerConfig().getStageListenerArn())
+              .stageListenerRuleArn(asgBlueGreenSwapServiceRequest.getAsgLoadBalancerConfig().getStageListenerRuleArn())
+              .stageTargetGroupArnList(
+                  asgBlueGreenSwapServiceRequest.getAsgLoadBalancerConfig().getStageTargetGroupArnsList())
```
[the remark this answers is not in the document]
Comment (lovish1999) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java]: hmm, ok

File: 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java (hunk 8 above)
Comment (lovish1999) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java]: hmm, yes removing them

File: 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java (hunk 7 above)
Comment (lovish1999) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java:138]: I was confused in the naming. Changing that.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java (hunk 6 above)
Comment (lovish1999) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java]: I was confused in the naming. Changing that.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java (hunk 5 above)
Comment (lovish1999) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/aws/asg/AsgBlueGreenSwapServiceStep.java]: I was confused in the naming. Changing that.

**Review by vitsafronovici — DISMISSED** [review]
> Reviewed 7 of 23 files at r1, 13 of 17 files at r2, 8 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42712)* status: 20 file(s) to review. 1 change requests. 8. unresolved discussions. 1 of 2 approvals obtained.

**Comment by arvind-choudhary-h** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 127 at r3](https://reviewable.io/reviews/harness/harness-core/42712#-NMSsyYX-aPur2Ml-a3q:-NMSsyYX-aPur2Ml-a3r:b-sb18m1) ([raw file](https://github.com/harness/harness-core/blob/6842240db9b03bc0af06d8daea803c3f5d5c50f8/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L127)):*
> ```Java
>                          .addHandler(AsgConfiguration,
>                              AsgConfigurationManifestRequest.builder()
>                                  .manifests(Arrays.asList(asgConfigurationContent))
> ```

how will the downsizing work? here, we are not providing any ASG name so it will default to ASG name in the request.?
Also, in the if/else block initial code till AsgSwapService is duplicate, let's merge it.

**Comment by arvind-choudhary-h** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 164 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkzsa-3EH0LauK7-a:-NMSwkBPDNp2P2ix55bI:bodeo0q) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L164)):*
<details><summary><i>Previously, lovish1999 (Lovish Bansal) wrote…</i></summary><blockquote>

if swapping fails, this will remain by default false. 
</blockquote></details>

Where is the second object of AsgBlueGreenSwapServiceResult being created?



_Code quote:_
```Java
AsgBlueGreenSwapServiceResult
```

**Comment by arvind-choudhary-h** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 164 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkzsa-3EH0LauK7-a:-NMSwt-B6fA9tfIPd2Jr:b-t3i2gy) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L164)):*
<details><summary><i>Previously, arvind-choudhary-h (Arvind Choudhary) wrote…</i></summary><blockquote>

Where is the second object of AsgBlueGreenSwapServiceResult being created?
</blockquote></details>

In the catch section, there is no AsgBlueGreenSwapServiceResult being created

**Review by arvind-choudhary-h — COMMENTED** [review]
> Reviewed 6 of 23 files at r1, 10 of 17 files at r2, 4 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42712)* status: Some of the checks are too old. 7 file(s) to review. 1 change requests. 8. unresolved discussions. 1 of 2 approvals obtained.

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42712)* status: Some of the checks are too old. 7 file(s) to review. 1 change requests. 9. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 164 at r1](https://reviewable.io/reviews/harness/harness-core/42712#-NLzkzsa-3EH0LauK7-a:-NMTnvC_CNCaIG6XGJUU:bc7x07d) ([raw file](https://github.com/harness/harness-core/blob/bbcd3e575948f3ea171151ac5f75a7530273ed73/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L164)):*
<details><summary><i>Previously, arvind-choudhary-h (Arvind Choudhary) wrote…</i></summary><blockquote>

In the catch section, there is no AsgBlueGreenSwapServiceResult being created
</blockquote></details>

we are just throwing an exception in that case
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 127 at r3](https://reviewable.io/reviews/harness/harness-core/42712#-NMSsyYX-aPur2Ml-a3q:-NMTmlPFClGLylWxxVxm:b-u7r9l) ([raw file](https://github.com/harness/harness-core/blob/6842240db9b03bc0af06d8daea803c3f5d5c50f8/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L127)):*
<details><summary><i>Previously, arvind-choudhary-h (Arvind Choudhary) wrote…</i></summary><blockquote>

how will the downsizing work? here, we are not providing any ASG name so it will default to ASG name in the request.?
Also, in the if/else block initial code till AsgSwapService is duplicate, let's merge it.
</blockquote></details>

we are providing the asgname in initial chain state.

**Comment by arvind-choudhary-h** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java` line 127 at r3](https://reviewable.io/reviews/harness/harness-core/42712#-NMSsyYX-aPur2Ml-a3q:-NMXXRly5OLo_SKvrWLB:b-5n1yjk) ([raw file](https://github.com/harness/harness-core/blob/6842240db9b03bc0af06d8daea803c3f5d5c50f8/930-delegate-tasks/src/main/java/io/harness/delegate/aws/asg/AsgBlueGreenSwapServiceCommandTaskHandler.java#L127)):*
<details><summary><i>Previously, lovish1999 (Lovish Bansal) wrote…</i></summary><blockquote>

we are providing the asgname in initial chain state. 
</blockquote></details>

Yeah, now it's fine -- earlier it was buggy. Also, make sure to merge these chains in next PRs.

**Review by arvind-choudhary-h — APPROVED** [review]
> Reviewed 1 of 23 files at r1, 1 of 17 files at r2, 3 of 8 files at r3, 22 of 22 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42712)* status: 22 file(s) to review. 1 change requests. 8. unresolved discussions. 1 of 2 approvals obtained.

**Comment by lovish1999** [comment]
trigger ti0

**Review by vitsafronovici — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42712#-:-NMXxbO2DdEXC9xEfJ7J:bnfp4nl)* status: 22 file(s) to review. 1 change requests. 8. unresolved discussions.

**Pull request merged**